sdk/identity/azure-identity/azure/identity/_credentials/azd_cli.py

```diff
@@ -0,0 +1,166 @@
+
+    @log_get_token("AzureDeveloperCliCredential")
+    def get_token(self, *scopes: str, **kwargs) -> AccessToken:
+        """Request an access token for `scopes`.
+
+        This method is called automatically by Azure SDK clients. Applications calling this method directly must
+        also handle token caching because this credential doesn't cache the tokens it acquires.
+
+        :param str scopes: desired scope for the access token. This credential allows only one scope per request.
+            For more information about scopes, see
+            https://learn.microsoft.com/azure/active-directory/develop/scopes-oidc.
+        :keyword str tenant_id: optional tenant to include in the token request.
+
+        :rtype: :class:`azure.core.credentials.AccessToken`
+
+        :raises ~azure.identity.CredentialUnavailableError: the credential was unable to invoke the Azure Developer CLI.
+        :raises ~azure.core.exceptions.ClientAuthenticationError: the credential invoked the Azure Developer CLI but didn't
+          receive an access token.
+        """
+
+        if not scopes:
+            raise ValueError("Missing scope in request. \n")
+
+        commandString = ' --scope '.join(scopes)
+        command = COMMAND_LINE.format(commandString)
+        tenant = resolve_tenant(
+            default_tenant=self.tenant_id,
+            additionally_allowed_tenants=self._additionally_allowed_tenants,
+            **kwargs
+        )
+        if tenant:
+            command += " --tenant-id " + tenant
+        output = _run_command(command)
+
+        token = parse_token(output)
+        if not token:
+            sanitized_output = sanitize_output(output)
+            raise ClientAuthenticationError(
+                message="Unexpected output from Azure Developer CLI: '{}'. \n".format(sanitized_output))
+
+        return token
+
+
+def parse_token(output):
+    """Parse to an AccessToken.
+
+    In particular, convert the "expiresOn" value to epoch seconds. This value is a naive local datetime as returned by
+    datetime.fromtimestamp.
+    """
+    try:
+        token = json.loads(output)
+        dt = datetime.strptime(token["expiresOn"], "%Y-%m-%dT%H:%M:%SZ")
+        expires_on = dt.timestamp()
+
+        return AccessToken(token["token"], int(expires_on))
+    except (KeyError, ValueError):
+        return None
+
+
+def get_safe_working_dir():
+    """Invoke 'azd' from a directory controlled by the OS, not the executing program's directory"""
+
+    if sys.platform.startswith("win"):
+        path = os.environ.get("SYSTEMROOT")
+        if not path:
+            raise CredentialUnavailableError(message="Azure Developer CLI credential expects a 'SystemRoot' environment variable")
+        return path
+
+    return "/bin"
+
+
+def sanitize_output(output):
+    """Redact tokens from CLI output to prevent error messages revealing them"""
+    return re.sub(r"\"token\": \"(.*?)(\"|$)", "****", output)
+
+
+def _run_command(command):
```

Comment:
Can we reuse the _run_command from azure_clie.py?

Reply:
Then we may need to port this fix https://github.com/Azure/azure-sdk-for-python/pull/27991/files. :)